File: leetcode/Isomorphic_Strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        data = {}
        used = set()
        for i in range(len(s)):
            if not data.get(s[i], 0):
                if t[i] in used:
                    return False
                data[s[i]] = t[i]
                used.add(t[i])
            elif data[s[i]]!= t[i]:
                return False
        return True
    

print(Solution.isIsomorphic(Solution, "badc", "baba"))

File: leetcode/Isomorphic_Strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        map<char, char> data;
        set<char> used;
        for (size_t i = 0; i < s.size(); i++)
        {  
            if (not data[s[i]])
            {
                if (used.contains(t[i]))
                    return false;
                data[s[i]] = t[i];
                used.insert(t[i]);
            }
            else if (data[s[i]] != t[i])
                return false;
            
        }
        return true;
        
    }
};